### ISSUE TITLE
Missing Egg File

### ISSUE THREAD

**Issue by trentnbauer** [metrogamelab/stardew-valley-mods#2]
G'day guys,
Looks like the egg file is missing from the repo - could you please to add it?
Thanks

**Comment by BaggyG-AU** [comment]
@xardoniak , did you get this working and if you did can you share an export of your egg?

**Comment by Just3itx** [comment]
i really need the egg 📦

**Comment by Jacob26502** [comment]
bump

**Comment by ObiWanCanOweMe** [comment]
bump

**Comment by jeremy-sidekicksolutions** [comment]
I hope metrogamelab is ok

**Comment by EnteSuesssauer15** [comment]
anyone made the egg themselves?

**Comment by XxCRSxX** [comment]
bump

**Comment by DNAniel213** [comment]
bump

**Comment by ianmuninio** [comment]
> anyone made the egg themselves?

I'm trying to make an egg and I'm halfway done. Right now, running with just pure docker takes a lot of CPU unit. I'm looking for alternatives or improvements to optimize it.

**Comment by DNAniel213** [comment]
bumping! need the egg!

**Comment by dyrkuwu** [comment]
bump lol

**Comment by jpotter23** [comment]
> > anyone made the egg themselves?
> 
> I'm trying to make an egg and I'm halfway done. Right now, running with just pure docker takes a lot of CPU unit. I'm looking for alternatives or improvements to optimize it.

You ever get this working and if so will you share the egg?

**Comment by jpotter23** [comment]
@ianmuninio 
> I'm trying to make an egg and I'm halfway done. Right now, running with just pure docker takes a lot of CPU unit. I'm looking for alternatives or improvements to optimize it.

Did you finish this? Willing to share? Even with high CPU usage, I'd like to see the egg in action, if possible ;)

**Comment by Bombombur** [comment]
@ianmuninio

> > anyone made the egg themselves?
> 
> I'm trying to make an egg and I'm halfway done. Right now, running with just pure docker takes a lot of CPU unit. I'm looking for alternatives or improvements to optimize it.

And update? Did you get it work?

**Comment by bUmmy1337** [comment]
maybe publish egg?

**Comment by paytah232** [comment]
Like you, I found this hoping for an egg but didn't get anything, so I forked this and had a crack myself. Actually took quite a lot of effort, and a lot of consulting AI, but seems to work for me, so give it a go and feel free to fork and update it if you need to.

https://github.com/paytah232/pterodactyl-server-stardew-valley

**Comment by XxCRSxX** [comment]
> Like you, I found this hoping for an egg but didn't get anything, so I forked this and had a crack myself. Actually took quite a lot of effort, and a lot of consulting AI, but seems to work for me, so give it a go and feel free to fork and update it if you need to.
> 
> https://github.com/paytah232/pterodactyl-server-stardew-valley

Impressive... will give it a go when I'm next messing about with stardew. Great to see a solution to this after all these years ha!

**Comment by paytah232** [comment]
> > Like you, I found this hoping for an egg but didn't get anything, so I forked this and had a crack myself. Actually took quite a lot of effort, and a lot of consulting AI, but seems to work for me, so give it a go and feel free to fork and update it if you need to.
> > https://github.com/paytah232/pterodactyl-server-stardew-valley
> 
> Impressive... will give it a go when I'm next messing about with stardew. Great to see a solution to this after all these years ha!

Yeah have a go - like @ianmuninio, the CPU usage is quite high. I don't understand enough about it to help bring that down (I'm sure someone could), but I run a beefy server and have lots of CPU power available, so it doesn't affect me at all. I'd expect it's because in order to run this you actually need to force a display (fails to launch otherwise), so you end up using CPU power to generate that on a headless server; unlike true dedicated servers which actually run headless. You're not actually running a dedicated server, you're running the game itself - your host is actually in the game and running a display, you've just been teleported off map so you're not visible. You'll see when you go to sleep that the host character also has to go to sleep, and the mods go through any menus etc for you - cool, but extremely wasteful.

**Comment by DNAniel213** [comment]
@paytah232 incredible!! It works for me. Definitely high CPU usage though

**Comment by paytah232** [comment]
@DNAniel213 @XxCRSxX - Some more AI, it gave me something to effectively limit FPS. I set it to 5, and it reduced the CPU usage by  ~4-6x, something like 600% usage to ~100-200%.

<img width="615" height="1140" alt="Image" src="https://github.com/user-attachments/assets/075c877c-db2f-481e-adfc-9ce48adf2d2c" />

I didn't play in game long enough to notice if it actually affects something in the game. So try the new egg variable and script. If you don't want to reinstall completely, you can manually add the variable and manually override the `stardew-valley-server.sh` file, or, just  replace the `stardew-valley-server.sh` file directly in your server, and change the 30 in this line to 5 (and back to 30 to disable).
`FPS="${RENDER_FPS:-30}"`

**Comment by Bombombur** [comment]
> [@DNAniel213](https://github.com/DNAniel213) [@XxCRSxX](https://github.com/XxCRSxX) - Some more AI, it gave me something to effectively limit FPS. I set it to 5, and it reduced the CPU usage by ~4-6x, something like 600% usage to ~100-200%.
> <img alt="Image" width="615" height="1140" src="https://private-user-images.githubusercontent.com/44075588/532742806-075c877c-db2f-481e-adfc-9ce48adf2d2c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgwNzE1NTEsIm5iZiI6MTc2ODA3MTI1MSwicGF0aCI6Ii80NDA3NTU4OC81MzI3NDI4MDYtMDc1Yzg3N2MtZGIyZi00ODFlLWFkZmMtOWNlNDhhZGYyZDJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTEwVDE4NTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5NWVkYWFmYjYxZTIwMjVjMDMwMWYzNzljYTE1ZGUzYzcxNTA3MzFkMzY5ODcwYzRlZjZmYWIwNDhhZDAyMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DEZLkJXMItRLYEPDNh7UEgiI0bLsxUwVZPoHgshZU_k">
> 
> I didn't play in game long enough to notice if it actually affects something in the game. So try the new egg variable and script. If you don't want to reinstall completely, you can manually add the variable and manually override the `stardew-valley-server.sh` file, or, just replace the `stardew-valley-server.sh` file directly in your server, and change the 30 in this line to 5 (and back to 30 to disable). `FPS="${RENDER_FPS:-30}"`

Helps only sometimes and for every event vnc is needed atm, seems something not correct with the mod or the ServerMod isn't able to run with SVE (collection) but that hasn't overall with the Egg. 

Just FYI.

**Comment by paytah232** [comment]
> > [@DNAniel213](https://github.com/DNAniel213) [@XxCRSxX](https://github.com/XxCRSxX) - Some more AI, it gave me something to effectively limit FPS. I set it to 5, and it reduced the CPU usage by ~4-6x, something like 600% usage to ~100-200%.
> > <img alt="Image" width="615" height="1140" src="https://private-user-images.githubusercontent.com/44075588/532742806-075c877c-db2f-481e-adfc-9ce48adf2d2c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgwNzE1NTEsIm5iZiI6MTc2ODA3MTI1MSwicGF0aCI6Ii80NDA3NTU4OC81MzI3NDI4MDYtMDc1Yzg3N2MtZGIyZi00ODFlLWFkZmMtOWNlNDhhZGYyZDJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTEwVDE4NTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5NWVkYWFmYjYxZTIwMjVjMDMwMWYzNzljYTE1ZGUzYzcxNTA3MzFkMzY5ODcwYzRlZjZmYWIwNDhhZDAyMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DEZLkJXMItRLYEPDNh7UEgiI0bLsxUwVZPoHgshZU_k">
> > I didn't play in game long enough to notice if it actually affects something in the game. So try the new egg variable and script. If you don't want to reinstall completely, you can manually add the variable and manually override the `stardew-valley-server.sh` file, or, just replace the `stardew-valley-server.sh` file directly in your server, and change the 30 in this line to 5 (and back to 30 to disable). `FPS="${RENDER_FPS:-30}"`
> 
> Helps only sometimes and for every event vnc is needed atm, seems something not correct with the mod or the ServerMod isn't able to run with SVE (collection) but that hasn't overall with the Egg.
> 
> Just FYI.

Yeah good to know, we haven't started our multilayer world yet, so I guess we'll have the same. I'll have to keep an eye on the mods to see if any update resolves that.

**Comment by Bombombur** [comment]
> > 
> > 
> > Helps only sometimes and for every event vnc is needed atm, seems something not correct with the mod or the ServerMod isn't able to run with SVE (collection) but that hasn't overall with the Egg.
> > Just FYI.
> 
> Yeah good to know, we haven't started our multilayer world yet, so I guess we'll have the same. I'll have to keep an eye on the mods to see if any update resolves that.

FYI: https://www.nexusmods.com/games/stardewvalley/collections/tckf0m is the collection we use/test.

**Comment by Bombombur** [comment]
> > > Helps only sometimes and for every event vnc is needed atm, seems something not correct with the mod or the ServerMod isn't able to run with SVE (collection) but that hasn't overall with the Egg.
> > > Just FYI.
> > 
> > 
> > Yeah good to know, we haven't started our multilayer world yet, so I guess we'll have the same. I'll have to keep an eye on the mods to see if any update resolves that.
> 
> FYI: https://www.nexusmods.com/games/stardewvalley/collections/tckf0m is the collection we use/test.

Seems someone wrote a fix for AutoHideHost already but was hard to find for me as i wasn't able to read it :D 

https://github.com/eaiu/patch-for-puppy-stardew-server which seems to fix mostly the issues i had with.
